// @serverless/enterprise-plugin/dist/lib/utils.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.upperFirst = upperFirst;
exports.pickResourceType = pickResourceType;
exports.API_GATEWAY_LOG_FORMAT = exports.LAMBDA_FILTER_PATTERN = exports.API_GATEWAY_FILTER_PATTERN = void 0;

function upperFirst(str) {
  return str.charAt(0).toUpperCase() + str.slice(1);
}

function pickResourceType(template, resourcesType) {
  var resources = [];

  for (var key in template.Resources) {
    var resource = template.Resources[key];

    if (resource.Type === resourcesType) {
      resources.push({
        key,
        resource
      });
    }
  }

  return resources;
}

var API_GATEWAY_FILTER_PATTERN = '"SLS_ACCESS_LOG"';
exports.API_GATEWAY_FILTER_PATTERN = API_GATEWAY_FILTER_PATTERN;
var LAMBDA_FILTER_PATTERN = '?"REPORT RequestId: " ?"SERVERLESS_ENTERPRISE"';
exports.LAMBDA_FILTER_PATTERN = LAMBDA_FILTER_PATTERN;
var API_GATEWAY_LOG_FORMAT = {
  requestTime: '$context.requestTime',
  requestId: '$context.requestId',
  apiId: '$context.apiId',
  resourceId: '$context.resourceId',
  resourcePath: '$context.resourcePath',
  path: '$context.path',
  httpMethod: '$context.httpMethod',
  status: '$context.status',
  authLatency: '$context.authorizer.integrationLatency',
  integrationLatency: '$context.integrationLatency',
  integrationStatus: '$context.integrationStatus',
  responseLatency: '$context.responseLatency',
  responseLength: '$context.responseLength',
  errorMessage: '$context.error.message',
  format: 'SLS_ACCESS_LOG',
  version: '1.0.0'
};
exports.API_GATEWAY_LOG_FORMAT = API_GATEWAY_LOG_FORMAT;
